neurovault/apps/statmaps/static/scripts/metadata-edit.js

```diff
@@ -0,0 +1,382 @@
+    NVMetadata.headers.splice(index, 0, {
+      name: name
+    });
+    NVMetadata.data.map(function (x) {
+      x.splice(index, 0, '');
+    });
+  }
+
+  function removeDataColumn(index) {
+    NVMetadata.headers.splice(index, 1);
+    NVMetadata.data.map(function (x) {
+      x.splice(index, 1);
+    });
+  }
+
+  function insertLocation(selection, key) {
+    if (key === 'insert_column_left') {
+      return selection[1];
+    } else {
+      return selection[3] + 1;
+    }
+  }
+
+  function updateSettings(hotInstance) {
+    hotInstance.updateSettings({
+      data: window.NVMetadata.data,
+      colHeaders: headerNames(window.NVMetadata.headers),
+      columns: columnSettings(window.NVMetadata.headers)
+    });
+  }
+
+  function handleInsertColumn(hotInstance, selection, key) {
+    hotInstance.deselectCell();
+
+    getNewColumnName(function (name) {
+      var index = insertLocation(selection, key);
+
+      insertDataColumn(index, name);
+      updateSettings(hotInstance);
+      hotInstance.render();
+    });
+  }
+
+  function handleRemoveColumn(hotInstance, selection) {
+    removeDataColumn(selection[1]);
+    updateSettings(hotInstance);
+    hotInstance.render();
+  }
+
+  $(document).ready(function () {
+
+    var container = document.getElementById('hot'),
+      hot = new Handsontable(container, {
+        data: window.NVMetadata.data,
+        colHeaders: headerNames(window.NVMetadata.headers),
+        columns: columnSettings(window.NVMetadata.headers),
+        rowHeaders: true,
+        height: getDefaultHeight(window.NVMetadata.data),
+        allowInsertRow: false,
+        minSpareRows: 0,
+        maxRows: window.NVMetadata.data.length,
+        contextMenu: {
+          callback: function (key) {
+            if (key === 'insert_column_left' ||
+              key === 'insert_column_right') {
+              handleInsertColumn(hot, hot.getSelected(), key);
+            } else if (key === 'remove_this_column') {
+              handleRemoveColumn(hot, hot.getSelected());
+            }
+          },
+          items: {
+            'insert_column_left': {
+              name: 'Insert column on the left',
+              disabled: function () {
+                return hot.getSelected()[1] === 0
+              }
+            },
+            'insert_column_right': {
+              name: 'Insert column on the right'
+            },
+            'hsep2': '---------',
+            'remove_this_column': {
+              name: 'Remove column',
+              disabled: function () {
+                return isFixedColumn(hot.getSelected()[1])
+              }
+            },
+            'hsep3': '---------',
+            'undo': {},
+            'redo': {}
+          }
+        },
+        cells: function (r, c, prop) {
+          if (c === 0) {
+            return {
+              readOnly: true
+            };
+          }
+        }
+      });
+
+    $('.btn-save-metadata').click(function (e) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['e' is defined but never used.](https://www.codacy.com/app/krzysztof-gorgolewski/NeuroVault_NeuroVault/file/825092758/issues/source?bid=2767031&fileBranchId=2767194#l350)